src/tgp.cpp

```diff
@@ -995,8 +995,8 @@ void GenerateTerrainPerlin()
 
 	/* First make sure the tiles at the north border are void tiles if needed. */
 	if (_settings_game.construction.freeform_edges) {
-		for (int y = 0; y < _height_map.size_y - 1; y++) MakeVoid(_height_map.size_x * y);
-		for (int x = 0; x < _height_map.size_x;     x++) MakeVoid(x);
```

Comment:
seems to me like _height_map could go away entirely, but that's for another time :)